templates/CRM/Group/Form/Search.tpl

```diff
@@ -234,15 +239,18 @@
                   'parent_is_' + parent_id,
                   'crm-row-child'
                 ];
+                if (!$.inArray("crm-smart-group", val.row_classes)) {
+                  smartGroupText = '';
+                }
```

Comment:
Try this instead:

```
                if ('DT_RowClass' in val) {
                  val.row_classes = val.row_classes.concat(val.DT_RowClass.split(' ').filter((item) => val.row_classes.indexOf(item) < 0));

                 // this ->
                  if (val.DT_RowClass.indexOf('crm-smart-group') == -1) {
                    smartGroupText = '';
                  }
                }
```

because the `val.row_classes`, set just above, will not have the `crm-smart-group` class. It gets set from the `DT_RowClass` in the condition right under.

Reply:
Awesome, thanks for the patch :)